minethd.cpp

```diff
@@ -33,7 +33,11 @@
 
 void thd_setaffinity(std::thread::native_handle_type h, uint64_t cpu_id)
 {
+#ifdef __MINGW64__
+	SetThreadAffinityMask(&h, 1ULL << cpu_id);
```

Comment:
> HANDLEs on Windows are usually passed by value, as you can see in the code on the other side of the #else.

Then it seem there's a slight difference in the SetThreadAffinityMask provided by MSVC and MinGW.

> My point is that the HANDLE = void* makes it possible for your code to fail silently (both at compile and run time) if you pass an address where a value is expected.

A pointer is expected in the MinGW version of SetThreadAffinityMask for the param **hThread**. As shown in the src x86_64-w64-mingw32\include\winnt.h (line 397 - 403), HANDLE is a typedef of void*. Thus a:
-  var, where var is T* var
-  &var, where var is T var

is expected in SetThreadAffinityMask (**HANDLE hThread**, NOT a value.

That is why there's a #ifdef guard specifically for MinGW64.

> My point is that the HANDLE = void* makes it possible for your code to fail silently (both at compile and run time) if you pass an address where a value is expected.

The code **only** get compiled for MinGW64 platforms only.
The implementation that expects values can fall back to the #else code

In conclusion, the best way to know is to try it out yourself, then you'll have surity.